pyk/src/pyk/rewrite.py

```diff
@@ -103,3 +159,40 @@ def combine_dicts(dict1: Optional[Mapping[K, V]], dict2: Optional[Mapping[K, V]]
 def combine_all(*dicts: Optional[Mapping[K, V]]) -> Optional[Dict[K, V]]:
     unit: Optional[Dict[K, V]] = {}
     return reduce(combine_dicts, dicts, unit)
+
+
+def rewrite(rule: Tuple[KInner, KInner], term: KInner) -> KInner:
+    """
+    Rewrite a given term at the top with the supplied rule.
+
+    :param rule: A rewrite rule of the form (lhs, rhs).
+    :param term: Term to rewrite.
+    :return: The term with the rewrite applied once at the top.
+    """
+    lhs, rhs = rule
+    subst = match(lhs, term)
+    if subst is not None:
+        return subst(rhs)
+    return term
+
+
+def rewrite_anywhere(rule: Tuple[KInner, KInner], term: KInner) -> KInner:
```

Comment:
Can we also make all these functions take as argument a `KRewrite` instead of a `Tuple[KInner,KInner]`?

Reply:
Maybe `Union[Tuple[KInner, KInner], KRewrite]`?

Reply:
Or let's just add them as convenience methods to `KRewrite`.

Reply:
Why `Union`? We're already updating all the downstream code-bases, and it's easy enough for people to pass a `KRewrite(...)` instead of a `Tuple`, since they're isomorphic.

It's just a matter of stricter typing.

Reply:
Alright, I'll add them to `KRewrite` as methods then.

Reply:
Hmmmm, but that requires `Subst` to be defined when `KRewrite` is defined, is that possible?

Reply:
But that would be cool! Just not worth mangling dependencies to achieve.

Reply:
> Hmmmm, but that requires Subst to be defined when KRewrite is defined

Yeah, good point. But we can merge `pyk.rewrite` into `pyk.kast`. Then we can maybe rename that module to `pyk.core`. Or is that too much?

Reply:
Done.